node/network/bitfield-distribution/src/lib.rs

```diff
@@ -147,65 +146,82 @@ impl BitfieldDistribution {
 	}
 
 	/// Start processing work as passed on from the Overseer.
-	async fn run<Context>(self, mut ctx: Context) -> SubsystemResult<()>
+	async fn run<Context>(self, mut ctx: Context)
 	where
 		Context: SubsystemContext<Message = BitfieldDistributionMessage>,
 	{
 		// work: process incoming messages from the overseer and process accordingly.
 		let mut state = ProtocolState::default();
 		loop {
-			let message = ctx.recv().await?;
+			let message = match ctx.recv().await {
+				Ok(message) => message,
+				Err(e) => {
+					warn!(target: LOG_TARGET, "Failed to receive a message from Overseer: {}, exiting", e);
```

Comment:
I think we should print this as a debug message, because it is legal to have the sender being closed when the node is closed

Reply:
```suggestion
					debug!(target: LOG_TARGET, "Failed to receive a message from Overseer: {}, exiting", e);
```